### PR TITLE
[11.0][ADD] stock_putaway_same_location

### PR DESCRIPTION
**Pull request by ghost**
### Summary
This module adds a new putaway strategy called "Previous location or empty one".
This put away strategy will select the destination location as the last used location known for a given product.

### TODO
- [ ] handle multiple warehouses
- [x] provide a description

### CODE REVIEW

File: stock_putaway_same_location/tests/test_same_location_putaway_strategy.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+class SameLocationPutawayCase(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # Common data
+        cls.location_stock = cls.env.ref('stock.stock_location_stock')
+        tested_putaway_strategy = cls.env['product.putaway'].create({
+            'name': 'Tested putaway strategy',
+            'method': 'previous/empty',
```
Comment (gurneyalex) [inline stock_putaway_same_location/tests/test_same_location_putaway_strategy.py]: this crashes in Travis:

ValueError: Wrong value for product.putaway.method: 'previous/empty'

File: stock_putaway_same_location/models/stock_previous_putaway_strat.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2018 by Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
```
Comment (gurneyalex) [inline stock_putaway_same_location/models/stock_previous_putaway_strat.py]: missing blank line => red travis

**Review by gurneyalex — CHANGES_REQUESTED** [review]

File: stock_putaway_same_location/models/product_putaway.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        domain = [
+            '|',
+            ('quant_ids.quantity', '=', 0.),
+            ('quant_ids', '=', False),
+            ('usage', '=', 'internal'),
+            ('location_id', 'child_of', root_location.id),
+        ]
+        if exclude_root:
+            domain.append(('id', '!=', root_location.id))
+        # FIXME is that efficient enough?
```
Comment (simahawk) [inline stock_putaway_same_location/models/product_putaway.py]: needs fix then?

**Review by simahawk — APPROVED** [review]
quick code review: LGTM.

**Review by gurneyalex — APPROVED** [review]

**Comment by jgrandguillaume** [comment]
Hi @arkostyuk ,


Thanks fro the PR, could you provide the description so we can merge it ?

Thanks,

Joël

**Review by max3903 — APPROVED** [review]

**Review by bodedra — APPROVED** [review]

**Pull request merged**